src/Microsoft.Health.Dicom.SqlServer/Features/Workitem/WorkitemSqlQueryGenerator.cs

```diff
@@ -0,0 +1,179 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Query;
+using Microsoft.Health.Dicom.Core.Features.Query.Model;
+using Microsoft.Health.Dicom.SqlServer.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.SqlServer.Features.Query;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Storage;
+
+namespace Microsoft.Health.Dicom.SqlServer.Features.Workitem
+{
+    internal class WorkitemSqlQueryGenerator : BaseSqlQueryGenerator
```

Comment:
Agree with this Will - I've extracted more common fields and methods into the base class and went through the unit tests to see what tests applied. Most of the existing tests are related to level hierarchy as they relate to extended query tags, or check data types which workitem doesn't use currently. I've added some tests, but I think your suggestion will require follow-up work.